### PR TITLE
Fix/style/layout responsivness/navigation UI

### PR DESCRIPTION
**Pull request by francelu**
# Responsive Layout: Adjust Padding and Typography in `Bottom Navigation Menu` and Top App `Bar`

## Description
This PR adjusts the padding and typography in the bottom navigation menu and top app bar to ensure consistent and responsive layouts across different screen sizes. It addresses subtask #144 of issue #82 by refining UI components for a better user experience and visual alignment.

## Changes
- Updated padding values for the bottom navigation menu and top app bar using responsive dimensions.
- Applied `labelSmall` typography to `BottomNavigationMenu`.
- Ensured consistent use of `MaterialTheme.dimens` and `MaterialTheme.typography` for padding and text styling.

#### Modified
- `BottomNavigationMenu.kt`: Adjusted padding and applied `labelSmall` typography.
- `TopAppBar.kt`: Updated padding for consistent layout.
- `Type.kt`: Added `labelLarge`.

## Dependencies Added
- None

## Testing
- Verified on multiple emulators to ensure consistent and responsive design across different screen sizes.

## Screenshots

| Screen              | Before `BottomNavigationBar` and `EditButton` | After `BottomNavigationBar` |
|---------------------|-------------------------------------|-------------------------------------|
| **Compact S**       | Before Compact S     | After Compact S      |
| **Compact M**       | Before Compact M     | After Compact M      |
| **Compact L**       | Before Compact L     | After Compact L      |
| **Medium (horizontal)** | Before Medium horizontal | After Medium horizontal |
| **Medium**          | Before Medium        | After Medium        |
| **Extended**        | Before Extended     | After Extended      |

| Screen              | Before `TopAppBar` and `GoBackButton` | After `TopAppBar` and `GoBackButton` |
|---------------------|-------------------------------------|-------------------------------------|
| **Compact S**       | Before Compact S     | After Compact S      |
| **Compact M**       | Before Compact M     | After Compact M      |
| **Compact L**       | Before Compact L     | After Compact L      |
| **Medium (horizontal)** | Before Medium horizontal | After Medium horizontal |
| **Medium**          | Before Medium        | After Medium        |
| **Extended**        | Before Extended     | After Extended      |

### CODE REVIEW

**Review by coaguila — APPROVED** [review]
Lgtm

File: app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt [hunk 1]
```diff
@@ -27,27 +28,24 @@ fun BottomNavigationMenu(
   NavigationBar(
       modifier =
           Modifier.fillMaxWidth()
-              .height(60.dp)
+              .wrapContentHeight()
               .testTag(BottomNavigationMenu.BOTTOM_NAVIGATION_MENU),
-      containerColor = MaterialTheme.colorScheme.surface,
+      containerColor = MaterialTheme.colorScheme.primaryContainer,
```
Comment (agonzalez-r) [inline app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt:33]: why did you change the colour?

**Review by agonzalez-r — APPROVED** [review]
Good implementation of the typography and padding. 
LGTM
I'm just curious about the change of colour of the bottom bar. Why did you make that choice?

File: app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt (hunk 1 above)
Comment (francelu) [reply app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt:33]: Because I couldn't see how "tall", the height of the bottom navigation.  
Colours will properly be handled in another task, don't worry :)
